tests/Unit/Infrastructure/Services/HttpRequestReadySessionTest.php

```diff
@@ -0,0 +1,102 @@
+use StepupReadId\Domain\ReadySession\Model\ReadySessionTimestamp;
+use StepupReadId\Domain\ReadySession\Model\ReadySessionTTL;
+use StepupReadId\Infrastructure\Services\HttpReadIDClientInterface;
+use StepupReadId\Infrastructure\Services\HttpRequestReadySession;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+class HttpRequestReadySessionTest extends TestCase
+{
+    /** @var HttpReadIDClientInterface */
+    private $httpClient;
+    /** @var ResponseInterface */
+    private $response;
+
+    protected function setUp(): void
+    {
+        $this->httpClient = $this->getMockBuilder(HttpReadIDClientInterface::class)->getMock();
+        $this->response   = $this->getMockBuilder(ResponseInterface::class)->getMock();
+
+        $this->httpClient
+            ->expects($this->exactly(1))
+            ->method('createReadySession')
+            ->with(ReadySessionTTL::MINIMUM_TTL)
+            ->willReturn($this->response);
+    }
+
+    public function testReceiveRequest(): void
+    {
+        $this->response
+            ->expects($this->exactly(2))
+            ->method('getStatusCode')
+            ->willReturn(Response::HTTP_CREATED);
+
+        $this->response
+            ->expects($this->exactly(1))
+            ->method('toArray')
+            ->willReturn([
+                HttpRequestReadySession::READY_SESSION_ID => '2e32c9a2-e34c-4f40-ac2f-dc47b67f3900',
+                HttpRequestReadySession::BASE_64_QR => 'data:image/png;base64,0',
+                HttpRequestReadySession::EXPIRY_TIMESTAMP => 1,
+                HttpRequestReadySession::JWK_TOKEN => 'j.w.t',
+            ]);
+
+        $requestReadySession = new HttpRequestReadySession($this->httpClient);
+        $readySession        = $requestReadySession->with(
+            ReadySessionTTL::fromInteger(ReadySessionTTL::MINIMUM_TTL)
+        );
+
+        $this->assertEquals(
+            ReadySession::create(
+                ReadySessionId::fromString('2e32c9a2-e34c-4f40-ac2f-dc47b67f3900'),
+                ReadySessionBase64Image::fromString('data:image/png;base64,0'),
+                ReadySessionJwtToken::fromString('j.w.t'),
+                ReadySessionTimestamp::fromInteger(1)
+            ),
+            $readySession
+        );
+    }
+
+    /**
+     * @expectedException \StepupReadId\Domain\ReadySession\Exception\RequestReadySessionAuthorizationException
+     */
+    public function testUnauthorizedRequest(): void
+    {
+        $this->response
+            ->expects($this->exactly(1))
+            ->method('getStatusCode')
+            ->willReturn(Response::HTTP_UNAUTHORIZED);
+
+        $requestReadySession = new HttpRequestReadySession($this->httpClient);
+        $requestReadySession->with(
+            ReadySessionTTL::fromInteger(ReadySessionTTL::MINIMUM_TTL)
+        );
+    }
+
+    /**
+     * @expectedException \StepupReadId\Domain\ReadySession\Exception\RequestReadySessionBadRequestException
```

Comment:
Using the `expectedException` annotations are discouraged. And will be removed from PHPUnit 8. Instead use `$this->expectException(Name::class)` in code.

https://github.com/sebastianbergmann/phpunit/issues/3332